packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1546,6 +1553,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this._perfSignalData.signalsLost = 0;
             this._perfSignalData.signalTimestamp = 0;
             this._perfSignalData.trackingSignalSequenceNumber = undefined;
+        } else {
+            assert(this.attachState === AttachState.Attached,
```

Comment:
should this change be part of this PR?

Reply:
Some tests failed with this change, and they were incorrectly "hacking" container runtime, so it was easier to catch wrongdoing by adding more checks like that. I can move it out, but 1-liners do not feel worth the effort :)

Reply:
Actually, it's also part of https://github.com/microsoft/FluidFramework/pull/11823 :)